### PR TITLE
Remove haystack

### PR DESCRIPTION
**Pull request by JoepdeJong**

### CODE REVIEW

File: ldb/views.py [hunk 1]
```diff
@@ -39,11 +37,29 @@ class ResultsView(TemplateView):
 
     def get_results(self):
         q = self.request.GET.get("q")
+        print(q)
```
Comment (jgadelange) [inline ldb/views.py:40]: print statement

File: ldb/views.py [hunk 2]
```diff
@@ -39,11 +37,29 @@ class ResultsView(TemplateView):
...
+                )
+                organization_filters &= Q(name__icontains=word)
+
+            # Return all persons
+            return list(
+                chain(
+                    # Filter person on firstname, surname, email and address
+                    Person.objects.filter(person_filters),
+                    Organization.objects.filter(organization_filters),
+                )
```
Comment (jgadelange) [inline ldb/views.py:62]: Hmmm... by converting this to a list Django will return all results from the database. Later we strip this by only returning the first 10 elements.

limiting both queries to 10 (by adding `[:10]` after the `.filter()`) could be a quickfix. However, then the `count` won't be correct anymore. 

Also: I see the count is achieved by using `len()` it is beter to `.count()` on the queryset. (maybe the counts should also be added to this method and returned separately?)

File: ldb/templates/ldb/results.html [hunk 3]
```diff
@@ -7,21 +7,19 @@ <h3 class="panel-title">
...
-            {% if result.object.ldap_username %}
-              <small><span class="label label-default">{{ result.object.ldap_username }}</span></small>
-            {% endif %}
-          </h4>
-          {% if result.object.entity_ptr.country %}
-            <p>{{ result.object.entity_ptr }}</p>
+      <a href="{{ result.get_absolute_url }}" class="list-group-item">
+        <h4 class="list-group-item-heading">
+          {{ result }}
+          {% if result.content_type == "ldb.organization" %}
```
Comment (jgadelange) [inline ldb/templates/ldb/results.html:13]: I don't think this `if` will ever be true. Did you check that?

File: ldb/views.py [hunk 4]
```diff
@@ -52,9 +67,9 @@ def get_context_data(self, **kwargs):
         results = self.get_results()
         context.update(
             {
-                "results": results[:10],
+                "results": results,
                 "count": len(results),
-                "remainder": len(results) - 10,
+                "remainder": len(results) - 20,
```
Comment (jgadelange) [inline ldb/views.py:72]: Deze klopt nu niet meer omdat het maximaal 20 gaat zijn 😅

File: ldb/views.py [hunk 5]
```diff
@@ -40,10 +38,27 @@ class ResultsView(TemplateView):
...
+            for word in q:
+                person_filters &= (
+                    Q(firstname__icontains=word)
+                    | Q(preposition__icontains=word)
+                    | Q(surname__icontains=word)
+                    | Q(ldap_username__icontains=word)
+                    | Q(street_name__icontains=word)
+                )
+                organization_filters &= Q(name__icontains=word)
+
```
Comment (jgadelange) [inline ldb/views.py:54]: Meer iets als dit? 
```
person_qs = Person.objects.filter(person_filters)
organization_qs = Organization.objects.filter(organization_filters)

count = person_qs.count() + organization_qs.count()
return list(chain(person_qs[:10], organization_qs[:10])), count
```

File: ldb/views.py (hunk 4 above)
Comment (JoepdeJong) [reply ldb/views.py:72]: ah zooo, snap m

**Review by julian9499 — APPROVED** [review]

**Pull request merged**